src/openlcb/NonAuthoritativeEventProducer.hxx

```diff
@@ -0,0 +1,207 @@
+    ///                       consumer identified.  The first unsigned parameter
+    ///                       represents the bit offset for the range and
+    ///                       the second bool parameter indicates the state as
+    ///                       true for valid and false for invalid
+    BitRangeNonAuthoritativeEventP(Node *node, uint64_t event_base_on,
+                  uint64_t event_base_off, uint32_t size,
+                  std::function<void(unsigned, bool)> state_callback = nullptr)
+        : node_(node)
+        , eventBaseOn_(event_base_on)
+        , eventBaseOff_(event_base_off)
+        , size_(size)
+        , stateCallback_(state_callback)
+    {
+        unsigned mask;
+
+        mask = EventRegistry::align_mask(&event_base_on, size);
+        EventRegistry::instance()->register_handler(
+            EventRegistryEntry(this, event_base_on), mask);
+
+        mask = EventRegistry::align_mask(&event_base_off, size);
+        EventRegistry::instance()->register_handler(
+            EventRegistryEntry(this, event_base_off), mask);
+    }
+
+    /// Destructor.
+    ~BitRangeNonAuthoritativeEventP()
+    {
+        EventRegistry::instance()->unregister_handler(this);
+    }
+
+    /// Queries consumer and acquires the current state of the bit.
+    ///
+    /// @param bit bit pair offset from the base event ID of the range,
+    ///            (0 <= bit < size)
+    /// @param writer object that will assist in the write transaction
+    /// @param done notifible to wakup when finished
+    void send_query_consumer(unsigned bit, WriteHelper *writer,
+                             BarrierNotifiable *done);
+
+    /// Requests the event associated with the current value of the bit to be
+    /// produced (unconditionally).
+    ///
+    /// @param bit is the offset of the bit to set (0 <= bit < size)
+    /// @param new_value is the new value of the bit
+    /// @param writer is the output flow to be used
+    /// @param done notifible to wakup when finished
+    void set(unsigned bit, bool new_value, WriteHelper *writer,
+             BarrierNotifiable *done);
+
+    /// Handle an incoming event.
+    ///
+    /// @param entry reference to this entry in the event registry
+    /// @param event event metadata
+    /// @param done notifible to wakup when finished
+    void handle_event_report(const EventRegistryEntry &entry,
+                             EventReport *event,
+                             BarrierNotifiable *done) override;
+
+    /// Handle an incoming consumer identified message.
+    ///
+    /// @param entry reference to this entry in the event registry
+    /// @param event event metadata
+    /// @param done notifible to wakup when finished
+    void handle_consumer_identified(const EventRegistryEntry &entry,
+                                    EventReport *event,
+                                    BarrierNotifiable *done) override;
+
+    /// Handle an incoming identify producer message.
```

Comment:
Actually, is it not both?  Which hander gets called for an Identify Events Addressed?